### PR TITLE
RUMM-2606 Battery Status Publisher

### PR DESCRIPTION
**Pull request by maxep**
### What and why?

Replace `BatteryStatusReader` with `BatteryStatusPublisher`.

### Review checklist
- [x] Feature or bugfix MUST have appropriate tests (unit, integration)
- [x] Make sure each commit and the PR mention the Issue number or JIRA reference
- [ ] Add CHANGELOG entry for user facing changes

### Custom CI job configuration (optional)
- [ ] Run unit tests
- [ ] Run integration tests
- [ ] Run smoke tests

### CODE REVIEW

File: Sources/Datadog/DatadogCore/DatadogCore.swift [hunk 1]
```diff
@@ -359,7 +359,8 @@ extension DatadogContextProvider {
         #endif
 
         #if os(iOS) && !targetEnvironment(simulator)
-        assign(reader: BatteryStatusReader(), to: \.batteryStatus)
+        subscribe(\.batteryStatus, to: BatteryStatusPublisher())
+        subscribe(\.isLowPowerModeEnabled, to: LowPowerModePublisher())
```
Comment (maxep) [inline Sources/Datadog/DatadogCore/DatadogCore.swift:363]: This was a miss for previous change.

File: Datadog/Datadog.xcodeproj/project.pbxproj [hunk 2]
```diff
@@ -6855,6 +6859,7 @@
 				ASSETCATALOG_COMPILER_APPICON_NAME = AppIcon;
 				CLANG_ENABLE_MODULES = YES;
 				CODE_SIGN_STYLE = Automatic;
+				DEVELOPMENT_TEAM = JKFCB4CN7C;
```
Comment (ncreated) [inline Datadog/Datadog.xcodeproj/project.pbxproj:6862]: There are 3 diffs on `DEVELOPMENT_TEAM` that need to be reverted (so it defaults to automatic team selection).

**Review by ncreated — APPROVED** [review]
Nice 🎯💪 

Found one minor to revert in `pbxproj` but everything else looks clear 👌.

File: Datadog/Datadog.xcodeproj/project.pbxproj (hunk 2 above)
Comment (maxep) [reply Datadog/Datadog.xcodeproj/project.pbxproj:6862]: oh good catch, thanks!

Comment (ncreated) [reply Datadog/Datadog.xcodeproj/project.pbxproj:6862]: Also, to test on device `DEVELOPMENT_TEAM` is needed 👍, but better way to pass this setting is through `Datadog.local.xcconfig` (gitignored):
```
DEVELOPMENT_TEAM[sdk=iphoneos*]=JKFCB4CN7C
```

Comment (maxep) [reply Datadog/Datadog.xcodeproj/project.pbxproj:6862]: `⌘C`/`⌘V` 😄  Thank you!!

**Pull request merged**